File: src/main/java/client/effects/user/AvatarOrientedEffect.java
package client.effects.user;

import client.effects.AbstractEffect;
import client.effects.EffectType;
import util.packet.PacketWriter;

public class AvatarOrientedEffect extends AbstractEffect {

    private final String path;

    public AvatarOrientedEffect(String path) {
        this.path = path;
    }

    @Override
    public EffectType getType() {
        return EffectType.AVATAR_ORIENTED;
    }

    @Override
    protected void encodeData(PacketWriter pw) {
        pw.writeMapleString(path);
        pw.writeInt(0);
    }
}

File: src/main/java/client/effects/user/AvatarOrientedEffect.kt
package client.effects.user

import client.effects.AbstractEffect
import client.effects.EffectType
import util.packet.PacketWriter

class AvatarOrientedEffect(val path: String) : AbstractEffect() {

    override val type = EffectType.AVATAR_ORIENTED

    override fun encodeData(pw: PacketWriter) {
        pw.writeMapleString(path)
        pw.writeInt(0)
    }
}